modules/stochastic_tools/include/samplers/LatinHypercubeSampler.h

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Sampler.h"
+
+class LatinHypercubeSampler;
+
+template <>
+InputParameters validParams<LatinHypercubeSampler>();
+/**
+ * A class used to perform Monte Carlo Sampling
+ */
+class LatinHypercubeSampler : public Sampler
+{
+public:
+  static InputParameters validParams();
+
+  LatinHypercubeSampler(const InputParameters & parameters);
+
+protected:
+  virtual void sampleSetUp() override;
+  virtual void sampleTearDown() override;
+  virtual Real computeSample(dof_id_type row_index, dof_id_type col_index) override;
+
+  /// Storage for distribution objects to be utilized
+  std::vector<Distribution const *> _distributions;
+
+  /// Distribution names
+  const std::vector<DistributionName> & _distribution_names;
+
+  /// Number of intervals for each distribution
+  const std::vector<unsigned int> & _num_bins_input;
+
+  /// Upper limit to probability for each distribution (use double b/c MooseRandom uses double)
+  const std::vector<double> & _upper_limits;
```

Comment:
Sampler inherits from MooseObject so it's tightly tied into MOOSE and MOOSE types. Is there any reason not to use `Real`?

Reply:
The values get sent directly to MooseRandom, which uses double. It seemed like I should keep it the same.